### PR TITLE
Pin official default images

### PR DESCRIPTION
**Pull request by 6543**
address https://github.com/woodpecker-ci/woodpecker/issues/1508#issuecomment-1368007494

### CODE REVIEW

File: docs/docs/30-administration/10-server-config.md [hunk 1]
```diff
@@ -203,7 +203,7 @@ Always use authentication to clone repositories even if they are public. Needed
 List of event names that will be canceled when a new pipeline for the same context (tag, branch) is created.
 
 ### `WOODPECKER_DEFAULT_CLONE_IMAGE`
-> Default: `woodpeckerci/plugin-git:latest`
+> Default is defined in [shared/constant/constant.go](https://github.com/woodpecker-ci/woodpecker/blob/release/v0.15/shared/constant/constant.go)
```
Comment (6543) [inline docs/docs/30-administration/10-server-config.md:206]:
```suggestion
> Default is defined in [shared/constant/constant.go](https://github.com/woodpecker-ci/woodpecker/blob/master/shared/constant/constant.go)
```

File: shared/constant/constant.go [hunk 2]
```diff
@@ -14,12 +14,13 @@
...
-	"plugins/docker",
-	"plugins/gcr",
-	"plugins/ecr",
-	"woodpeckerci/plugin-docker",
-	"woodpeckerci/plugin-docker-buildx",
+	"docker.io/plugins/docker:20.14.0",
+	"docker.io/plugins/gcr:20.14.0",
+	"docker.io/plugins/ecr:20.14.0",
+	"docker.io/woodpeckerci/plugin-docker-buildx:2.1.0",
+	// "docker.io/woodpeckerci/plugin-docker",
```
Comment (anbraten) [inline shared/constant/constant.go:23]: I am not sure if this list support tags and if we should pin those images as they are used for checking the permissions and not for fetching images.

Comment (lafriks) [reply shared/constant/constant.go:23]: This will be hard to manage to have to count all allowed tags for each image

Comment (lafriks) [reply shared/constant/constant.go:23]: Also I don't see how that improves security as if you can push new tag you can also push malicious tag over existing version tag. For security reasons we would have to pin to image tag sha256 than

File: shared/constant/constant.go [hunk 3]
```diff
@@ -14,12 +14,13 @@
...
 package constant
 
+// PrivilegedPlugins can be changed by 'WOODPECKER_ESCALATE' at runtime
 var PrivilegedPlugins = []string{
 	"plugins/docker",
 	"plugins/gcr",
 	"plugins/ecr",
-	"woodpeckerci/plugin-docker",
 	"woodpeckerci/plugin-docker-buildx",
+	// "woodpeckerci/plugin-docker",
```
Comment (lafriks) [inline shared/constant/constant.go:23]:
```suggestion
```

**Review by lafriks — APPROVED** [review]

**Pull request merged**